user/views.py

```diff
@@ -1,4 +1,92 @@
+            data = json.loads(request.body)
+
+            first_name              = data['first_name']
+            last_name               = data['last_name']
+            email                   = data['email']
+            password                = data['password']
+            confirm_password        = data['confirm_password']
+            receive_communications  = data['receive_communications']
+
+            if not first_name:
+                return JsonResponse({'MESSAGE' : 'FIRST_NAME_IS_REQUIRED'}, status=400)
+
+            if not last_name:
+                return JsonResponse({'MESSAGE' : 'LAST_NAME_IS_REQUIRED'}, status=400)
+
+            if not email:
+                return JsonResponse({'MESSAGE' : 'EMAIL_IS_REQUIRED'}, status=400)
+
+            if not password:
+                return JsonResponse({'MESSAGE' : 'PASSWORD_IS_REQUIRED'}, status=400)
+
+            if not confirm_password:
+                return JsonResponse({'MESSAGE' : 'CONFRIM_PASSWORD_IS_REQUIRED'}, status=400)
+
+            if not re.search(email_regax, email):
+                return JsonResponse({'MESSAGE' : 'INVALID_EMAIL'}, status=400)
+
+            if Users.objects.filter(email=email).exists():
+                return JsonResponse({'MESSAGE' : 'EXIST_USER'}, status=401)
+
+            if not re.search(password_regax, password):
+                return JsonResponse({'MESSAGE' : 'INVALID_PASSWORD'}, status=400)
+
+            if not password == confirm_password:
+                return JsonResponse({'MESSAGE' : 'PASSWORD_UNMATCH'}, status=400)
+
+            save_password    = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+            
+            Users.objects.create(
+                first_name             = first_name,
+                last_name              = last_name,
+                email                  = email,
+                password               = save_password,
+                receive_communications = receive_communications
+            )
+     
+            return JsonResponse({'MESSAGES' : 'SUCCESS'}, status=200)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE' : 'KEY_ERRORS'}, status=400)
+
+class LoginView(View):
+    def post(self, request):
+        try:
+            data           = json.loads(request.body)
+            
+            email          = data['email']
+            input_password = data['password']
+            users          = Users.objects.get(email=email)
+            
+            if not email:
+                return JsonResponse({'MESSAGE' : 'EMAIL_IS_REQUIRED'}, status=400)
+
+            if not input_password:
+                return JsonResponse({'MESSAGE' : 'PASSWORD_IS_REQUIRED'}, status=400)
+            
+            if bcrypt.chechpw(input_password.encode('utf-8'), users.password.encode('utf-8')):
+                token = jwt.encode({'user-id' : users.id}, SECRET, ALGORITHM)
+
+                return JsonResponse({token : 'token'}, status=200)
```

Comment:
token이 어떤 token인지 정확히 명시해주는게 좋습니다.
나중에 refresh_token등 다양한 토큰을 관리해야될 수 있습니다.
```suggestion
                return JsonResponse({access_token : 'token'}, status=200)
```

Reply:
헉 몰랐습니다 말씀대로 수정할게요 !